website/docs/r/opc_database_service_instance.html.markdown

```diff
@@ -0,0 +1,168 @@
+---
+layout: "opc"
+page_title: "Oracle: opc_database_service_instance"
+sidebar_current: "docs-opc-resource-service-instance"
+description: |-
+  Creates and manages a service instance in an OPC identity domain.
+---
+
+# opc\_database\_service\_instance
+
+The ``opc_database_service_instance`` resource creates and manages a service instance in an OPC identity domain.
+
+## Example Usage
+
+```hcl
+resource "opc_database_service_instance" "default" {
+  name        = "service-instance-1"
+  description = "This is a description for an service instance"
+  edition = "EE_EP"
+  level = "PAAS"
+  shape = "oc1m"
+  subscription_type = "HOURLY"
+  version = "12.2.0.1"
+  vm_public_key = "an ssh public key"
+  parameter {
+    admin_password = "Test_String7"
+    backup_destination = "NONE"
+    sid = "ORCL"
+    usable_storage = 15
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Service Instance.
+
+* `edition` - (Required) Database edition for the service instance. Possible values are SE, EE, EE_HP, or EE_EP.
+
+* `level` - (Required) Service level for the service instance. Possible values are BASIC or PAAS.
+
+* `shape` - (Required) Desired compute shape. Possible values are oc3, oc4, oc5, oc6, oc1m, oc2m, oc3m, or oc4m.
+
+* `subscription_type` - (Required) Billing unit. Possible values are HOURLY or MONTHLY.
+
+* `version` - (Required) Oracle Database software version; one of: 12.2.0.1, 12.1.0.2, or 11.2.0.4.
+
+* `vm_public_key` - (Required) Public key for the secure shell (SSH). This key will be used for authentication when connecting to the Database Cloud Service instance using an SSH client.
+
+* `parameter` - (Optional) Additional configuration for a service instance. This set is required if level is PAAS. Parameter is documented below.
+
+* `description` - (Optional) A description of the Service Instance.
+
+Parameter supports the following:
+
+* `admin_password` - (Required) Password for Oracle Database administrator users sys and system. The password must meet the following requirements: Starts with a letter. Is between 8 and 30 characters long. Contains letters, at least one number, and optionally, any number of these special characters: dollar sign ($), pound sign (#), and underscore ( _ )
```

Comment:
Oh, I misread. Their docs don't say anything about rejecting other characters. I just think this is added security. I'll add it to another test to be sure